p2p/switch.go

```diff
@@ -522,6 +529,10 @@ func (sw *Switch) addPeer(pc peerConn) error {
 
 	// Avoid self
 	if sw.nodeKey.ID() == peerID {
+		// overwrite current addr to avoid dialing ourselves again
+		// it means original nodePublicAddr was different from public one
```

Comment:
possible attack: send us some IP with our ID to prevent us from ever connecting to it

Reply:
The ID is authenticated so they could only do this attack if they have our privKey

Reply:
I thought we derive ID from pubkey, which is known and ... public

Reply:
Sure but when we do the AuthEnc handshake, they have to prove they control the privkey relating to the pubkey. They can try to connect with us claiming any ID they want, but as soon as we do the SecretHandshake, we will know for sure if they actually control that ID. That's how we use ID to prevent man-in-the-middle!

Reply:
OK, thank you for explaining